pkg/generator/ingress_translator.go

```diff
@@ -182,20 +185,34 @@ func (translator *IngressTranslator) translateIngress(ctx context.Context, ingre
 			return nil, nil
 		}
 
-		var virtualHost *route.VirtualHost
+		if len(ingress.Spec.TLS) != 0 {
+			tlsRoutes = append(tlsRoutes, routes...)
```

Comment:
This line (Copy `routes` to `tlsRoutes`) will not work as https://github.com/knative-sandbox/net-kourier/pull/619/  needs the condition - routes and tlsRoutes are not same when Redirect option.